sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -0,0 +1,595 @@
+    {
+
+      //get available segments from druidSchema
+      Map<String, ConcurrentSkipListMap<DataSegment, SegmentMetadataHolder>> getSegmentMetadataInfo = druidSchema.getSegmentMetadataInfo();
+      final Map<DataSegment, SegmentMetadataHolder> availableSegmentMetadata = new HashMap<>();
+      for (ConcurrentSkipListMap<DataSegment, SegmentMetadataHolder> val : getSegmentMetadataInfo.values()) {
+        availableSegmentMetadata.putAll(val);
+      }
+      final Iterator<Entry<DataSegment, SegmentMetadataHolder>> availableSegmentEntries = availableSegmentMetadata.entrySet()
+                                                                                                                  .iterator();
+
+      //get published segments from coordinator
+      final JsonParserIterator<DataSegment> metadataSegments = getMetadataSegments(
+          druidLeaderClient,
+          jsonMapper,
+          responseHandler
+      );
+
+      Set<String> availableSegmentIds = new HashSet<>();
+      final FluentIterable<Object[]> availableSegments = FluentIterable
+          .from(() -> availableSegmentEntries)
+          .transform(val -> {
+            try {
+              if (!availableSegmentIds.contains(val.getKey().getIdentifier())) {
+                availableSegmentIds.add(val.getKey().getIdentifier());
+              }
+              return new Object[]{
+                  val.getKey().getIdentifier(),
+                  val.getKey().getDataSource(),
+                  val.getKey().getInterval().getStart(),
+                  val.getKey().getInterval().getEnd(),
+                  val.getKey().getSize(),
+                  val.getKey().getVersion(),
+                  val.getKey().getShardSpec().getPartitionNum(),
+                  val.getValue().getNumReplicas(),
+                  val.getValue().getNumRows(),
+                  val.getValue().isPublished(),
+                  val.getValue().isAvailable(),
+                  val.getValue().isRealtime(),
+                  jsonMapper.writeValueAsString(val.getKey())
+              };
+            }
+            catch (JsonProcessingException e) {
+              log.error(e, "Error getting segment payload for segment %s", val.getKey().getIdentifier());
+              throw new RuntimeException(e);
+            }
+          });
+
+      final FluentIterable<Object[]> publishedSegments = FluentIterable
+          .from(() -> metadataSegments)
+          .transform(val -> {
+            try {
+              if (availableSegmentIds.contains(val.getIdentifier())) {
+                return null;
+              }
+              return new Object[]{
+                  val.getIdentifier(),
+                  val.getDataSource(),
+                  val.getInterval().getStart(),
+                  val.getInterval().getEnd(),
+                  val.getSize(),
+                  val.getVersion(),
+                  val.getShardSpec().getPartitionNum(),
+                  0l,
+                  -1l,
```

Comment:
I strongly feel we should add `numRows` to segment metadata, but it would be a different issue.

Reply:
hm, we can get `numRows` for the segment metadata from `DruidSchema`, but not for published segments we get from coordinator. Yes, agree that would be useful.